lib/PuppeteerSharp/NavigatorWatcher.cs

```diff
@@ -82,6 +82,17 @@ internal class NavigatorWatcher
         #endregion
 
         #region Private methods
+
+        private void OnFrameDetached(object sender, FrameEventArgs e)
+        {
+            var frame = e.Frame;
+            if (_frame == frame)
+            {
+                // TODO terminating callback

```

Comment:
@kblok I'm not really clear on how to translate this part here. Also, do you have any guidance on how to translate[ this test](https://github.com/GoogleChrome/puppeteer/blob/cf8c62e835b565feb9d19639a3832a59d9b07aca/test/frame.spec.js#L63)?

Thanks

Reply:
Here you should do something like this:
```suggestion
                _terminationTaskWrapper.TrySetException(new PuppeteerException("Navigating frame was detached"));
```

Regarding that test. This is how I would code it:
```cs
[Fact]
public async Task ShouldRejectWhenFrameDetaches()
{
    await Page.GoToAsync(TestConstants.ServerUrl + "/frames/one-frame.html");
    Server.SetRoute("/empty.html", context => Task.Delay(10000));
    var navigationTask = Page.Frames[1].GoToAsync(TestConstants.EmptyPage);
    await Server.WaitForRequest("/empty.html");
    await Page.QuerySelectorAsync("iframe").EvaluateFunctionAsync("frame => frame.remove()");
    var exception = await Assert.ThrowsAsync<PuppeteerException>(navigationTask);
    Assert.Equal("Navigating frame was detached", exception.Message);
}
```